lib/codelog/clis/interactive.rb

```diff
@@ -0,0 +1,54 @@
+require 'thor'
+
+module Codelog
+  module CLIs
+    class Interactive < Thor
+      include Thor::Actions
+
+      no_commands do
+        def initialize
+          @sections = YAML.load_file('changelogs/template.yml').keys
+        end
+
+        def ask_for_changes
+          change_type = ask_for_type
+          say "\nType the entries for \e[33m#{change_type}\e[0m(ENTER to stop):"
+          result_hash = { change_type => build_array_response }
+          if yes? "\nWould you like to add a new log(Y|N)?"
+            result_hash.merge!(ask_for_changes) do |_, changes, changes_to_be_added|
```

Comment:
Maybe we should think a little more about this recursion, for me its not clear enough when I should call `ask_for_changes` vs `build_array_response`, what do you think?